File: app/src/test/java/com/mgaetan89/showsrage/fragment/ShowsSectionFragment_GetCommandTest.java
package com.mgaetan89.showsrage.fragment;

import com.google.gson.Gson;
import com.mgaetan89.showsrage.model.Show;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ShowsSectionFragment_GetCommandTest {
	@Parameterized.Parameter(1)
	public String command;

	@Parameterized.Parameter(0)
	public List<Show> shows;

	@Test
	public void getCommand() {
		assertThat(ShowsSectionFragment.getCommand(this.shows)).isEqualTo(this.command);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		Gson gson = new Gson();

		return Arrays.asList(new Object[][]{
				{null, ""},
				{Collections.emptyList(), ""},
				{Collections.singletonList(gson.fromJson("{indexerid: 123}", Show.class)), "show.stats_123"},
				{Arrays.asList(
						gson.fromJson("{indexerid: 123}", Show.class),
						null,
						gson.fromJson("{indexerid: 456}", Show.class)
				), "show.stats_123|show.stats_456"},
				{Arrays.asList(
						gson.fromJson("{indexerid: 123}", Show.class),
						gson.fromJson("{indexerid: 456}", Show.class),
						gson.fromJson("{}", Show.class)
				), "show.stats_123|show.stats_456"},
				{Arrays.asList(
						gson.fromJson("{indexerid: 123}", Show.class),
						null,
						gson.fromJson("{}", Show.class)
				), "show.stats_123"},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/ShowsSectionFragment_GetCommandTest.kt
package com.mgaetan89.showsrage.fragment

import com.mgaetan89.showsrage.model.Show
import com.mgaetan89.showsrage.network.SickRageApi
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ShowsSectionFragment_GetCommandTest {
    @Parameterized.Parameter(1)
    var command: String? = null

    @Parameterized.Parameter(0)
    var shows: List<Show>? = null

    @Test
    fun getCommand() {
        assertThat(ShowsSectionFragment.getCommand(this.shows)).isEqualTo(this.command)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any?>> {
            val gson = SickRageApi.gson

            return listOf(
                    arrayOf<Any?>(null, ""),
                    arrayOf<Any?>(emptyList<Any>(), ""),
                    arrayOf<Any?>(listOf(gson.fromJson("{indexerid: 123}", Show::class.java)), "show.stats_123"),
                    arrayOf<Any?>(listOf(gson.fromJson("{indexerid: 123}", Show::class.java), null, gson.fromJson("{indexerid: 456}", Show::class.java)), "show.stats_123|show.stats_456"),
                    arrayOf<Any?>(listOf(gson.fromJson("{indexerid: 123}", Show::class.java), gson.fromJson("{indexerid: 456}", Show::class.java), gson.fromJson("{}", Show::class.java)), "show.stats_123|show.stats_456"),
                    arrayOf<Any?>(listOf(gson.fromJson("{indexerid: 123}", Show::class.java), null, gson.fromJson("{}", Show::class.java)), "show.stats_123")
            )
        }
    }
}
